cfg/qt.cfg

```diff
@@ -2458,6 +2458,180 @@
       <not-uninit/>
     </arg>
   </function>
+  <!-- QRegExp https://doc.qt.io/qt-5/qregexp.html -->
+  <!-- QString QRegExp::cap(int nth = 0) const -->
+  <function name="QRegExp::cap">
+    <noreturn>false</noreturn>
+    <returnValue type="QString"/>
+    <use-retval/>
+    <leak-ignore/>
+    <const/>
+    <arg nr="1" direction="in" default="0">
+      <not-uninit/>
+      <not-bool/>
+      <valid>0:</valid>
+    </arg>
+  </function>
+  <!-- int QRegExp::captureCount() const -->
+  <function name="QRegExp::captureCount">
+    <noreturn>false</noreturn>
+    <returnValue type="int"/>
+    <use-retval/>
+    <leak-ignore/>
+    <const/>
+  </function>
+  <!-- QStringList QRegExp::capturedTexts() const -->
+  <function name="QRegExp::capturedTexts">
+    <noreturn>false</noreturn>
+    <returnValue type="QStringList"/>
+    <use-retval/>
+    <leak-ignore/>
+    <const/>
+  </function>
+  <!-- Qt::CaseSensitivity QRegExp::caseSensitivity() const -->
+  <function name="QRegExp::caseSensitivity">
+    <noreturn>false</noreturn>
+    <returnValue type="Qt::CaseSensitivity"/>
+    <use-retval/>
+    <leak-ignore/>
+    <const/>
+  </function>
+  <!-- QString QRegExp::errorString() const -->
+  <function name="QRegExp::errorString">
+    <noreturn>false</noreturn>
+    <returnValue type="QString"/>
+    <use-retval/>
+    <leak-ignore/>
+    <const/>
+  </function>
+  <!-- QString QRegExp::escape(const QString &str) // static -->
+  <function name="QRegExp::escape">
```

Comment:
@danmar Does it make sense to mark `static` functions with `<const/>` and/or even with `<pure/>` (at least in some cases like this where obviously no global variable is changed)?

Reply:
yes it does. Then we should be able to detect cases like : `if (QRegExp::escape("a") == "x" && QRegExp::escape("a") == "y")`.. if it's pure/const is used we should be able to report that the condition is always false.

Reply:
Ok. I add `<pure/>` to this functions. `<const/>` is not necessary then, correct?
Would it make sense to add the possibility to add `<static/>` for functions in the library configurations?